doc/governance.rst

```diff
@@ -20,95 +20,107 @@ the project community.
 Roles And Responsibilities
 ==========================
 
+We distinguish between contributors, core contributors, and the technical
+committee. A key distinction between them is their voting rights: contributors
+have no voting rights, whereas the other two groups all have voting rights,
+as well as permissions to the tools relevant to their roles.
+
 Contributors
 ------------
 
 Contributors are community members who contribute in concrete ways to the
 project. Anyone can become a contributor, and contributions can take many forms
 – not only code – as detailed in the :ref:`contributors guide <contributing>`.
+There is no process to become a contributor: once somebody contributes to the
+project in any way, they are a contributor.
+
+Core Contributors
+-----------------
+
+All core contributor members have the same voting rights and right to propose
+new members to any of the roles listed below. In order to join one of the roles
+bellow. Their membership is represented as being an organization member on the
+scikit-learn `GitHub organization
+<https://github.com/orgs/scikit-learn/people>`_.
+
+They are also more than welcome to join our `monthly core contributor meetings
+<https://github.com/scikit-learn/administrative/tree/master/meeting_notes>`_.
+
+New members can be nominated by any existing member. Once they have been
+nominated, there will be a vote by the current core contributors. Voting on new
+members is one of the few activities that takes place on the project's private
+mailing list. While it is expected that most votes will be unanimous, a
+two-thirds majority of the cast votes is enough. The vote needs to be open for
+at least 1 week.
+
+Contributors that have not contributed to the project, corresponding to their
+role, in the past 12 months will be asked if they want to become emeritus
+members and recant their rights until they become active again. The list of
+members, active and emeritus (with dates at which they became active) is public
+on the scikit-learn website.
```

Comment:
I agree with having a mechanism like this one. Yet, who is responsible to check for contributors' inactivity?

Reply:
we have checked this periodically in the past, it hasn't been an issue. (if it ain't broken, why fix it)

Reply:
Currently, there are people listed as part of the project (especially as Core-Contributors) which have not been active for months or years. To you, should we do something about it?

Reply:
We have checked it periodically in the past. @amueller did a check recently, and we updated the list. IMO there's no action needed there, and even if there is, I don't think it's in the context of this PR. We also did a cleanup of users having access to pypi recently. I don't see that as a governance issue.